src/Microsoft.AspNetCore.Dispatcher/EndpointSelector.cs

```diff
@@ -8,5 +8,10 @@ namespace Microsoft.AspNetCore.Dispatcher
     public abstract class EndpointSelector
     {
         public abstract Task SelectAsync(EndpointSelectorContext context);
+
+        public virtual Task Initialize(IEndpointCollectionProvider endpointProvider)
```

Comment:
When/why did this become async? We need this to be synchronous because it's going to happen under a lock